### ISSUE TITLE
Constrain choice of End Date on Task creation UI to the end date of the Event

### ISSUE THREAD

**Issue by mgmccarthy** [HTBox/allReady#1022]
When going to create a new Task, the user has to choose a Start Date and an End Date:
![image](https://cloud.githubusercontent.com/assets/2428993/17022853/64dc0104-4f1e-11e6-9bba-1c842f2d808a.png)

The user is left to pick the start and end date on their own that looks like it defaults to DateTime.UtcNow or DateTimeOffset.UtcNow?

Either way, the user is allowed to pick any end date using the calendar control
![image](https://cloud.githubusercontent.com/assets/2428993/17022892/999423cc-4f1e-11e6-8a52-4ee81bf939ae.png)

only after being allowed to choose an end date that is after the end date of the campaign and clicking save does the user receive and error that the "End date cannot be later than the event end date 12/31/2015"
![image](https://cloud.githubusercontent.com/assets/2428993/17022801/1c536d78-4f1e-11e6-9cd2-a38eabfa2024.png)

Couple things we can do:
1. put the Event's end date on the UI so the user knows what the date range is that's acceptable without having to rely on an error to tell them.
2. even better, put default dates in the calendar for start and end date that are both within the range of the Event's end date. This is more work, but will guide the user to make the right choice by not allowing them to choose an start of end date outside the allowable range.

**Comment by VishalMadhvani** [comment]
I like option two.
I'm not sure which date picker we are using but they generally have minvalue/maxvalue properties for this exact use case.

Another thing.. if i'm not mistaken a task can not span across multiple days.
So in reality we need three fields here:
Date: which must be between event,startdate and event.enddate
Start Time.
End Time.

**Comment by mgmccarthy** [comment]
@VishalMadhvani, "if i'm not mistaken a task can not span across multiple days."... I think that is correct. I'll defer to @stevejgordon to see if that indeed is true. He's done a whole bunch of work lately around Tasks and Events

**Comment by stevejgordon** [comment]
I think it's only itinerary task which is single date bounded. While there was some text that said all taks where single day bounded, but on one of the standups I checked and rally events can be longer.

@tonysurma might be able to confirm I didn't dream this / or whether business rules have changed

**Comment by stevejgordon** [comment]
I put in logic to check that an itinerary task should be no longer than a day

**Comment by tonysurma** [comment]
@stevejgordon is correct only itinerary tasks are constrained to a single day

**Comment by DanKline** [comment]
I've noticed that the date populates briefly, like events used to, but then disappears.  It's probably the same issue,  I'm looking for it, but as neither an MVC or EF7 ninja, I don't have much confidence that I'll find it.
